### ISSUE TITLE
FAQs Warning Box

### ISSUE THREAD

**Issue by Biuwa** [hackforla/tdm-calculator#1279]
### Overview
To prevent admins from mistakenly deleting FAQs categories and questions, we need a warning dialog box that asks them to be sure before deleting 

### Action Items

- [x] create mockups for the warning box that quotes the text of the question/answer that the admin is deleting, and also includes a leaving the page warning box
- [x] Get stakeholders' approval

**Comment by Biuwa** [comment]
**John:**
For the question deletion warning dialog, it could be designed like the one for deleting projects:

![image](https://user-images.githubusercontent.com/98936028/204122262-1ad6cd37-77ce-4228-a886-0b8b945382c4.png)

Or, if you have an improved design, we should probably update the project deletion warning dialog to match.

**Comment by Biuwa** [comment]
@IMHieuVo :
I have updated mockups of the Warning Box per our 11/2 meeting (Jon noted that there is an existing Warning Box we could match).

Mockup A - Match. Match the existing warning box and the background fade as closely as possible while retaining the new language.

Mockup B - Dark Fade. Match the warning box but have the background be a darker fade.

Mockup C - New Style. Match elements of the warning boxes but have the look and feel of the newer style.

Mockup A - Match
![image](https://user-images.githubusercontent.com/98936028/204122410-76e2e66d-ae07-43c3-af5d-97a99211fc37.png)

Mockup B - Dark Fade
![image](https://user-images.githubusercontent.com/98936028/204122428-0087aedf-5df7-44fb-9518-dba003d9999a.png)

Mockup C - New Style
![image](https://user-images.githubusercontent.com/98936028/204122441-fcf54c93-605e-43d6-80c7-ab693f81f8b9.png)

**Comment by Biuwa** [comment]
**Syria:**
Team has selected Mockup C as the preferred style. @IMHieuVo

Per mtg 11/9, iterations needed:
-Add quoted text of question/answer that the user is deleting in the warning box
-To avoid accidental deletion, move focus to/highlight the "cancel" option

**Comment by Biuwa** [comment]
@IMHieuVo:
@Biuwa @SMR9220

Per the 11/9 meeting, the team has chosen Mockup C (Centered Text warning box with dark fade background). The latest iteration includes:

Quoting the text of the question/answer that the user is deleting in the warning box.
To avoid accidental deletion, move focus to/highlight the "cancel" option
Includes a Leaving-the-page warning box
Link: https://www.figma.com/file/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=7299%3A25362&t=0ztwOC3PNJ8aFPGN-1

Question Warning Box
![image](https://user-images.githubusercontent.com/98936028/204122594-b6bf7f88-0654-45a1-bd14-8a710fb3295b.png)

**Leaving warning box:**
![image](https://user-images.githubusercontent.com/98936028/204122611-138c370a-f881-42ef-b4b5-cf0e96083c98.png)

**Comment by Biuwa** [comment]
@IMHieuVo 
Below is the update of the warning boxes per accessibility considerations. The red text on the white background does not seem to pass the color contrast checker for accessibility so I opted to make the text black.

Option A) Uses the main red color as an outline
Option B) Uses bold white text on a dark red background.

https://www.figma.com/file/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=7419%3A25419&t=79Ho5XdabKSIQwbM-1

FAQ Question Warning Box, Option 1A - red outline
![image](https://user-images.githubusercontent.com/98936028/204122654-57ae499c-4523-4a05-a308-0a753a82fe91.png)

FAQ Question Warning Box, Option 1B - Red background
![image](https://user-images.githubusercontent.com/98936028/204122660-15723eab-096e-49ca-a7a8-fe5d8c3edb22.png)

FAQ Leaving Warning Box, Option 2A - Red outline
![image](https://user-images.githubusercontent.com/98936028/204122688-880c5cd1-fdbb-4d43-a00a-e466431dcf65.png)

FAQ Leaving Warning Box, Option 2B - Red background
![image](https://user-images.githubusercontent.com/98936028/204122699-41e9fa56-d07c-4828-ad3e-d75864e7fba6.png)

**Comment by Biuwa** [comment]
@ExperimentsInHonesty:
@IMHieuVo In the above post, you have several different warning messages, but two different color reds. Please explain why two different reds (the orange red and the more brown red).

**Comment by Biuwa** [comment]
@IMHieuVo:
@ExperimentsInHonesty
Short answer: Accessibility.
Long answer: The orange-red color is one of the 'core' colors of the site but it is a terrible color to use with white/gray in terms of accessibility. It fails both versions: white text on orange-red background and orange-red text on white background. In option A, we can still use the orange-red color because it is not directly interacting with the text. In option B, I had to go with a darker red color to pass the accessibility contrast.
![image](https://user-images.githubusercontent.com/98936028/204122830-d4dfef9e-0df1-4858-b425-6f68e47b34a9.png)

**Comment by Biuwa** [comment]
@IMHieuVo is it possible to have these design options in the different colors you used? For example, the red outline option will be in the 3 different shades of red you used?

Stakeholders having to choose between different designs in 3 different shades of red might be difficult. They might like a particular design, but may not like the color used, or may like a color and not the design.

**Comment by IMHieuVo** [comment]
@Biuwa 
Below is the preview of the expanded options per your request. Option A and Option D are the original. With only subtle differences, isn't this a bit much for the shareholders? 

![Frame 8](https://user-images.githubusercontent.com/117132709/204466300-2074186f-acc2-40f1-a68d-aba6eec4f762.png)

**Comment by ExperimentsInHonesty** [comment]
### Notes from 2023-01-10 meeting with stakeholder
- Approved by stakeholders, ready for dev (check warning box audit for color)

**Comment by IMHieuVo** [comment]
@Biuwa It is weird that the stakeholder chose Option 1B but wants the colors of Option 1A. The two options are identical except for the color red. It would make more sense to choose Option 1A from the start. Regardless, I've updated the warning box. 

- Update the red color to #E46247. 
- Add drop shadows (In my review of the warning box by @Noushie for issue 1289, there is both a drop shadow for the warning box and buttons. I matched it since the mockup was chosen)

![FAQ Question Warning Box - Option 1B - Update dropshadows](https://user-images.githubusercontent.com/117132709/212499072-3991ed65-2b5c-4dbe-8d1d-914f0b4837ea.png)

Figma Link: https://www.figma.com/file/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=7755%3A25886&t=gWyyDb2KkYR2Xj6W-1
